gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspConfigurations.kt

```diff
@@ -86,6 +86,16 @@ class KspConfigurations(private val project: Project) {
             // 1.6.0: decorateKotlinProject(project.kotlinExtension)?
             decorateKotlinProject(project.extensions.getByName("kotlin") as KotlinProjectExtension, project)
         }
+        // Create sourceSet-specific KSP configurations for the case when the KotlinBaseApiPlugin is applied instead
+        // of the KotlinBasePluginWrapper (e.g., when AGP's built-in Kotlin support is enabled).
```

Comment:
It's possible that they would both be applied, but there's currently no use case for it.

If they are both applied, this will still work because of the `maybeCreate` in `createConfigurations`.